src/main/java/Weeny/Weeny.java

```diff
@@ -0,0 +1,181 @@
+
+                case "todo":
+                    validateTodoInput(input);
+                    Task todoTask = new Todo(input.substring(5).trim());
+                    taskList.addTask(todoTask);
+                    return ui.printTaskAddedMessage(todoTask, taskList.size());
+
+                case "event":
+                    validateEventInput(input);
+                    Task eventTask = new Events(parser.extractEventName(input),
+                            parser.extractEventTimes(input)[0],
+                            parser.extractEventTimes(input)[1]);
+                    taskList.addTask(eventTask);
+                    return ui.printTaskAddedMessage(eventTask, taskList.size());
+
+                case "deadline":
+                    validateDeadlineInput(input);
+                    Task deadlineTask = new Deadlines(parser.extractDeadlineName(input),
+                            parser.extractDeadlineTime(input));
+                    taskList.addTask(deadlineTask);
+                    return ui.printTaskAddedMessage(deadlineTask, taskList.size());
+
+                case "delete":
+                    int deleteIndex = parser.extractEndNumber(input) - 1;
+                    validateIndex(deleteIndex, taskList.size(), "delete");
+                    Task removedTask = taskList.getTask(deleteIndex);
+                    taskList.deleteTask(deleteIndex);
+                    return ui.showTaskDeletedMessage(removedTask, taskList.size());
+
+                case "find":
+                    String keyWord = input.substring(5);
+                    return ui.showSearchResult(taskList.findTask(keyWord));
+
+                default:
+                    throw new UnsupportedOperationException("Unknown command");
+                }
+            } catch (IllegalArgumentException | IndexOutOfBoundsException | UnsupportedOperationException e) {
+                return ui.showError(e.getMessage());
+            }
+        }
+        storage.saveTask("./Data/TaskList.txt", taskList.getTasks());
+        return ui.showGoodbyeMessage();
+    }
+
+    /**
+     * Checks if the index is within the task list range.
+     *
+     * @param index The index to check.
+     * @param size The size of the task list.
+     * @param action The action being performed (e.g., "mark", "delete").
+     * @throws IndexOutOfBoundsException if the index is out of range.
+     */
+    protected static void validateIndex(int index, int size, String action) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Invalid index for " + action + " action.");
+        }
+    }
+
+    /**
+     * Validates input for a "todo" command.
+     *
+     * @param input The command input.
+     * @throws IllegalArgumentException if the input is too short.
+     */
+    private static void validateTodoInput(String input) {
+        if (input.length() <= 5) {
```

Comment:
What do you think about defining constants for the different length constraints?